### PR TITLE
feat: add nextjs implementation

### PR DESCRIPTION
**Pull request by LionyxML**
Adds NextJS implementation.

Bare bones for now, no optimization.

Haven't added results yet as in my machine it looks suspicious:

```bash
$ wrk -c 64 -d 30s http://localhost:4000/10
Running 30s test @ http://localhost:4000/10
  2 threads and 64 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    36.44ms   85.17ms   1.52s    98.25%
    Req/Sec     1.21k   342.10     1.57k    80.47%
  72195 requests in 30.06s, 8.54MB read
Requests/sec:   2401.87
Transfer/sec:    290.85KB
```

### CODE REVIEW

File: NodeJS/nextjs/next.config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/** @type {import('next').NextConfig} */
```
Comment (reu) [inline NodeJS/nextjs/next.config.js:1]: Is this really necessary? (I really don't know what this does hahaha)

**Comment by reu** [comment]
Thanks a lot! Could you add a Dockerfile also? You could use the other NodeJS implementations as a start.

File: NodeJS/nextjs/next.config.js (hunk 1 above)
Comment (LionyxML) [reply NodeJS/nextjs/next.config.js:1]: Yeah, really odd. This is part of the nextjs boilerplate.

As I understand it, this JSDoc helps provide type information and documentation for the configuration object, making it easier to work with and ensuring type safety in the code (although this is a js file).

Without it my editor fails to suggest completion.

More here: https://nextjs.org/docs/pages/api-reference/next-config-js

**Comment by LionyxML** [comment]
Dockerfile added.

**Pull request merged**